entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/generator/provider/DefaultValueHandlerProvider.kt

```diff
@@ -0,0 +1,25 @@
+package godot.entrygenerator.generator.provider
+
+import org.jetbrains.kotlin.builtins.KotlinBuiltIns
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.types.typeUtil.isEnum
+
+object DefaultValueHandlerProvider {
+    fun provideDefaultValueHandler(
+        propertyDescriptor: PropertyDescriptor,
+        bindingContext: BindingContext
+    ): RegistrationValuesHandler {
+        return when  {
+            KotlinBuiltIns.isInt(propertyDescriptor.type)
+                || KotlinBuiltIns.isInt(propertyDescriptor.type)
+                || KotlinBuiltIns.isLong(propertyDescriptor.type)
+                || KotlinBuiltIns.isFloat(propertyDescriptor.type)
+                || KotlinBuiltIns.isDouble(propertyDescriptor.type)
+                || KotlinBuiltIns.isBoolean(propertyDescriptor.type)
+                || KotlinBuiltIns.isString(propertyDescriptor.type) -> PrimitiveRegistrationValuesHandler(propertyDescriptor, bindingContext)
+            propertyDescriptor.type.isEnum() -> EnumRegistrationValuesHandler(propertyDescriptor, bindingContext)
+            else -> TODO()
```

Comment:
Maybe add a reason in todo ? like "Only primitives can be registered for now" ?

Reply:
I don't think this is necessary bacause we will have implemented all supported before the first beta release